crates/astria-composer/tests/blackbox/composer.rs

```diff
@@ -54,6 +54,43 @@ async fn tx_from_one_rollup_is_received_by_sequencer() {
     .expect("mocked sequencer should have received a broadcast message from composer");
 }
 
+#[tokio::test]
+async fn collector_restarts_after_exit() {
+    // Spawn a composer with a mock sequencer and a mock rollup node
+    // Initial nonce is 0
+    let test_composer = spawn_composer(&["test1"]).await;
+    tokio::time::timeout(
+        Duration::from_millis(100),
+        test_composer.setup_guard.wait_until_satisfied(),
+    )
+    .await
+    .expect("setup guard failed");
+
+    // get rollup node
+    let rollup_node = test_composer.rollup_nodes.get("test1").unwrap();
+    // abort the rollup node. The collector should restart after this abort
+    rollup_node.cancel_subscriptions().unwrap();
+
+    // wait for the collector to restart
+    tokio::time::sleep(Duration::from_millis(100)).await;
+
+    // the collector will be restarted now, we should be able to send a tx normally
+    let expected_rollup_ids = vec![RollupId::from_unhashed_bytes("test1")];
+    let mock_guard =
+        mount_broadcast_tx_sync_mock(&test_composer.sequencer, expected_rollup_ids, vec![0]).await;
+    test_composer.rollup_nodes["test1"]
+        .push_tx(Transaction::default())
+        .unwrap();
+
+    // wait for 1 sequencer block time to make sure the bundle is preempted
```

Comment:
Add a comment that you add 1000ms to make sure the websocket subscription to the collector comes back up